common/persistence/serialization/blob.go

```diff
@@ -223,16 +224,83 @@ func ChecksumToBlob(checksum *persistencespb.Checksum) (commonpb.DataBlob, error
 	return proto3Encode(checksum)
 }
 
-func ChecksumFromBlob(b []byte, proto string) (*persistencespb.Checksum, error) {
+func ChecksumFromBlob(binary []byte, encoding string) (*persistencespb.Checksum, error) {
 	result := &persistencespb.Checksum{}
-	err := proto3Decode(b, proto, result)
+	err := proto3Decode(binary, encoding, result)
 	if err != nil || result.GetFlavor() == enumsspb.CHECKSUM_FLAVOR_UNSPECIFIED {
 		// If result is an empty struct (Flavor is unspecified), replace it with nil, because everywhere in the code checksum is pointer type.
 		return nil, err
 	}
 	return result, nil
 }
 
+func QueueMessageToBlob(message *persistencespb.QueueMessage) (commonpb.DataBlob, error) {
+	// TODO change ENCODING_TYPE_JSON to ENCODING_TYPE_PROTO3
+	return encode(message, enumspb.ENCODING_TYPE_JSON)
+}
+
+func QueueMessageFromBlob(binary []byte, encoding string) (*persistencespb.QueueMessage, error) {
+	result := &persistencespb.QueueMessage{}
+	return result, decode(binary, encoding, result)
+}
+
+func QueueMetadataToBlob(metadata *persistencespb.QueueMetadata) (commonpb.DataBlob, error) {
+	// TODO change ENCODING_TYPE_JSON to ENCODING_TYPE_PROTO3
+	return encode(metadata, enumspb.ENCODING_TYPE_JSON)
+}
+
+func QueueMetadataFromBlob(binary []byte, encoding string) (*persistencespb.QueueMetadata, error) {
+	result := &persistencespb.QueueMetadata{}
+	return result, decode(binary, encoding, result)
+}
+
+func encode(
+	object interface{},
```

Comment:
```suggestion
	object proto.Message,
```

and change `proto3Encode` to accept `proto.Message` also, and then change:
```
data, err := m.Marshal()
```
to 
```
	data, err := proto.Marshal(m)
```
there.